### ISSUE TITLE
Using extensions with asciidoctorj 1.6.1

### ISSUE THREAD

**Issue by carlosmunoz** [asciidoctor/asciidoctorj#764]
I'm using asciidoctorj in an osgi environment and and I'm attempting to create a custom IncludeProcessor. Without any extensions everything works fine, but as soon as I register the processor I get the following stack trace:

```java
java.util.NoSuchElementException
	at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:365)
	at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	at org.asciidoctor.extension.BaseProcessor.<init>(BaseProcessor.java:36)
	at org.asciidoctor.extension.IncludeProcessor.<init>(IncludeProcessor.java:15)
	at org.asciidoctor.extension.IncludeProcessor.<init>(IncludeProcessor.java:11)
```

I tried to trace the problem but it seems the  BaseProcessor class has changed already in the latest snapshot (removing a FIX ME line which is probably where this comes from). I'm specifically registering the extension as a class instance, so the processor's lifecycle is managed externally.

**Comment by carlosmunoz** [comment]
Further experiments with the latest snapshot version (built locally) do confirm that the problem is fixed by the change in the BaseProcessor class. I guess my question is when can we have an official release? As I see it Java extensions might be broken in their current state.

**Comment by robertpanzer** [comment]
Yes, you're right, the changes in the BaseProcessor are not released yet.
The current master cannot be released as a 1.6.x version.
We'll have to move this particular fix into the asciidoctorJ-1.6.x branch and release that.

**Comment by carlosmunoz** [comment]
Further experiments with the latest snapshot version (built locally) do confirm that the problem is fixed by the change in the BaseProcessor class. I guess my question is when can we have an official release? As I see it Java extensions might be broken in the most current release.

**Comment by carlosmunoz** [comment]
Thanks @robertpanzer 
I Did a bit of digging and it seems that the relevant change is in fa8fd995c468cb6b49ea22194785f43f5dd31692 (though I'm not sure this change is not impacted or impacts other ones).

I assume you are not planning on releasing 1.7 any time soon then :smile:

**Comment by robertpanzer** [comment]
I'll try to do a 1.6.2 this week.

**Comment by robertpanzer** [comment]
Should be fixed with 1.6.2 that I just released.
Please retest and open if you still experience problems.

**Comment by carlosmunoz** [comment]
This works @robertpanzer 
Thanks heaps for the quick response.
